metrics.nim

```diff
@@ -941,10 +941,15 @@ when defined(metrics):
         var heapSizes: array[10, (cstring, int)]
         for data in dumpHeapInstances():
           var smallest = 0
-          for i in 1..<heapSizes.len:
+          var dedupe = false
+          for i in 0..<heapSizes.len:
+            if heapSizes[i][0] == data.name:
```

Comment:
consider:

```
Future[void]: 10
OtherType: 100
... other types 10x: 100
Future[void]: 95
```
in the above case, the sum of all `Future[void]` qualify for inclusion but none does individually - this is worth pointing out in a comment at least, as solving it is a bit more tricky given the constraints of not being allowed to allocate memory here (the solution would involve a larger array probably, to decrease the probability of data loss)

Reply:
Ah, good point. That's quite tricky. Truth be told I'm really bad at this kind of algorithmic stuff. I'm sure someone else could figure out a better solution.

Reply:
Maybe this could be avoided by first putting all data from `dumpHeapInstances` into one array and sorting it by names first?

Reply:
Ah, but we don't know in advance the number of entries `dumpHeapInstances` returns.